src/GafferModule/ValuePlugBinding.cpp

```diff
@@ -63,6 +63,31 @@ std::string repr( const ValuePlug *plug )
 	return ValuePlugSerialiser::repr( plug );
 }
 
+bool isSetToDefault( ValuePlug *plug )
```

Comment:
This is definitely the right thing to do given the current implementation of `ValuePlug::isSetToDefault()`, but I wonder if the implementation is wrong? Do we really expect `isSetToDefault()` to trigger a compute? If a compute is necessary, it implies that there may not be one true value for the plug (instead there's potentially a set of context-varying values). Should `isSetToDefault()` return `false` without doing a compute in this case?

I think there's a good case for just merging as-is to avoid having to think about the behaviour change right now, but I'd at least be interested in your thoughts on it...

Reply:
I think I agree with you on both counts - isSetToDefault should probably always return false when computation is required to get the plug value - but making this change would require a full audit of everywhere where it's used.  In a quick search, I'm not sure what CDL should be doing, and Reference::setInputsAndValues feel like it should maybe be doing something a bit different - I almost wonder if it should copy across the actual values stored on the plug as well as the input, since usually plugs do remember the value they were set to as well as the connection.

I also agree that the right thing for the moment is probably to merge this as-is to avoid deadlocks while we think about changing behaviour.